public/app/core/history/richHistoryLocalStorageUtils.ts

```diff
@@ -0,0 +1,82 @@
+import { RichHistoryQuery } from '../../types';
+import { omit } from 'lodash';
+import { SortOrder } from '../utils/richHistoryTypes';
+
+/**
+ * Temporary place for local storage specific items that are still in use in richHistory.ts
+ *
+ * Should be migrated to RichHistoryLocalStorage.ts
+ */
+
+export const createRetentionPeriodBoundary = (days: number, isLastTs: boolean) => {
+  const today = new Date();
+  const date = new Date(today.setDate(today.getDate() - days));
+  /*
+   * As a retention period boundaries, we consider:
+   * - The last timestamp equals to the 24:00 of the last day of retention
+   * - The first timestamp that equals to the 00:00 of the first day of retention
+   */
+  const boundary = isLastTs ? date.setHours(24, 0, 0, 0) : date.setHours(0, 0, 0, 0);
+  return boundary;
+};
+
+export function filterQueriesByTime(queries: RichHistoryQuery[], timeFilter: [number, number]) {
+  return queries.filter(
+    (q) =>
+      q.ts < createRetentionPeriodBoundary(timeFilter[0], true) &&
+      q.ts > createRetentionPeriodBoundary(timeFilter[1], false)
+  );
+}
+
+export function filterQueriesByDataSource(queries: RichHistoryQuery[], listOfDatasourceFilters: string[]) {
+  return listOfDatasourceFilters && listOfDatasourceFilters.length > 0
```

Comment:
```suggestion
  return listOfDatasourceFilters.length > 0
```
`listOfDatasourceFilters` should always be an array (according to the types)

Reply:
Fixed in 46cd5482300ab83b25f156b12b79e7a981ac39b1